app/src/main/java/communi/dog/aplicatiion/RegisterActivity.java

```diff
@@ -184,20 +184,26 @@ private boolean idDoubleUser(EditText id) {
     @Override
     protected void onSaveInstanceState(@NonNull Bundle outState) {
         super.onSaveInstanceState(outState);
-        outState.putString("userID", id.getText().toString());
-        outState.putString("userEmail", emailAddress.getText().toString());
-        outState.putString("userPass1", pass1.getText().toString());
-        outState.putString("userPass2", pass2.getText().toString());
-        outState.putString("userName", userName.getText().toString());
+        // as part of fix for issue 19 - no longer needed
+        // because every time we enter this screen we take the data from firebase
+
+//        outState.putString("userID", id.getText().toString());
+//        outState.putString("userEmail", emailAddress.getText().toString());
+//        outState.putString("userPass1", pass1.getText().toString());
+//        outState.putString("userPass2", pass2.getText().toString());
+//        outState.putString("userName", userName.getText().toString());
     }
 
     @Override
     protected void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
         super.onRestoreInstanceState(savedInstanceState);
-        id.setText(savedInstanceState.getString("userID"));
-        emailAddress.setText(savedInstanceState.getString("userEmail"));
-        pass1.setText(savedInstanceState.getString("userPass1"));
-        pass2.setText(savedInstanceState.getString("userPass2"));
-        userName.setText(savedInstanceState.getString("userName"));
+        // as part of fix for issue 19 - no longer needed
+        // because every time we enter this screen we take the data from firebase
+
+//        id.setText(savedInstanceState.getString("userID"));
+//        emailAddress.setText(savedInstanceState.getString("userEmail"));
+//        pass1.setText(savedInstanceState.getString("userPass1"));
+//        pass2.setText(savedInstanceState.getString("userPass2"));
+//        userName.setText(savedInstanceState.getString("userName"));
```

Comment:
I don't understand how issue19 is related to the RegisterActivity , it seems as if you where confused the ProfilePageActivity with the RegisterActivity .
furthermore, I don't agree with you that there is no need to save the current state of the EditTexts in the ProfilePageActivity, because if the user is in edit mode, the content of the EditTexts wont be the same as the field's of the current user and therefore it is essential to save the old editTexts content.
**Conclusion**:
* RegisterActivity - please revert your changes
* ProfilePageActivity - if you don't agree with me, please explain me why so we can get to an agreement about this issue.

Ido.